portal/src/graphql/portal/ForgotPasswordSettings.tsx

```diff
@@ -61,13 +77,66 @@ function constructAppConfigFromState(
       screenState.resetCodeExpirySeconds
     );
 
-    // TODO: update email template
-    // TODO: update sms template
+    if (
```

Comment:
Template items comes from config, which might not be empty. If user cleared an existing template (changed to empty string), we should remove the corresponding element? In current implementation we only append?

As are doing exactly same thing for three templates, I think we can abstract this to a function? something like

```typescript
updateTemplateItems(screenState, initialScreenState, key, templateItems) {
  // early return if no changes 
  if (initialScreenState[key] === screenState[key]) {
    return
  }
  // append if content non-empty and corresponding template item not exist
  ...
  // remove if content empty and corresponding template item exist
  ...
}
```

Reply:
@louischan-oursky Should we remove template item or simply set content to empty string?

Reply:
Remove. So that default value can be restored